pkg/apis/app/v1alpha1/kogitobuild_types.go

```diff
@@ -0,0 +1,236 @@
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
+	// +optional
+	DisableIncremental bool `json:"disableIncremental,omitempty"`
+
+	// Environment variables used during build time
+	// +listType=atomic
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Build Env Variables"
+	// +optional
+	Envs []corev1.EnvVar `json:"envs,omitempty"`
+
+	// Information about the git repository where the Kogito Service source code resides.
+	// Ignored for binary builds
+	// +optional
+	GitSource GitSource `json:"gitSource,omitempty"`
+
+	// Which runtime Kogito Runtime base image to use when building the Kogito Service.
+	// If "BuildImage" is set, this value is ignored by the operator.
+	// Default value: quarkus
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Runtime"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:label"
+	// +optional
+	// +kubebuilder:validation:Enum=quarkus;springboot
+	Runtime RuntimeType `json:"runtime,omitempty"`
+
+	// WebHooks secrets for source to image builds based on Git repositories (Remote Sources)
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	WebHooks []WebhookSecret `json:"webHooks,omitempty"`
+
+	// Native indicates if the Kogito Service built should be compiled to run on native mode when Runtime is Quarkus (Source to Image build only).
+	// For more information, see https://www.graalvm.org/docs/reference-manual/aot-compilation/.
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Native Build"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
+	Native bool `json:"native,omitempty"`
+
+	// Resources Requirements for builder pods.
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Maven Mirror URL to be used during source-to-image builds (Local and Remote) to considerably increase build speed
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Maven Mirror URL"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:label"
+	MavenMirrorURL string `json:"mavenMirrorURL,omitempty"`
```

Comment:
I don't think this setting copes with private Maven repositories, are we planning to support using private repositories in the future? It would be nice to create a ticket for this feature.

Reply:
Why not? You would set the public Maven URL to your private "group" repository, which groups all your enterprise repos. Not sure if I understood your question. Maybe elaborate a little bit more in format of story and open a JIRA?